### DISCUSSION TITLE
Marimo suddenly not starting on Ubuntu

### DISCUSSION THREAD

**Discussion by MrHenning** [marimo-team/marimo#778] [Q&A] [answered]
Hi, since yesterday I can no longer start marimo on Ubuntu...
E.g. running `marimo tutorial intro` does nothing at all.
Before it was running perfectly fine, and I don't see what has changed in the meantime.
I *can* run marimo by running it directly from Python rather than the command line:
```python
from marimo._server.model import SessionMode
from marimo._server.start import start

start(filename=None, mode=SessionMode.EDIT, development_mode=False, quiet=False, include_code=False, headless=True, port=None, host='localhost', watch=False)
```

Does anyone have an idea how I can investigate this??

**Comment by akshayka** [comment]
Uh oh! @MrHenning, can you run `marimo env` at the command line and paste the output here?

**Reply by akshayka** [reply]
Additionally, can you paste the output you see at the terminal when you run `marimo tutorial intro`? Or do you not see anything?

You can also try running with debug logs enabled: `marimo -l debug tutorial intro`

**Reply by MrHenning** [reply]
@akshayka all 3 commands give no output at all. I had already reinstalled marimo (using `mamba`). At some point I installed marimo as a PWA in Edge on Ubuntu, but I could still launch it after that.
I also tried to remove all traces of marimo (the mamba/conda package & environment, caches etc. from marimo and Edge just to be sure, but no change...).
I must have changed something to break it, but I can't figure out what...

**Reply by akshayka** [reply]
Thanks for the extra context, couple more questions.

What version of marimo are you using? If you install an older version, does it work?

If you install marimo from PyPI instead of conda forge, does it work?

**Reply by MrHenning** [reply]
I tried both `conda` and `pip`
I have 
```shell
marimo --version
0.2.5
```
(this does return something in the command line!)
and I also tried downgrading to `0.1.18`.

**Reply by akshayka** [reply]
It's very curious that version `0.1.18` does not work -- that's a very old version, and if it didn't work I wouldn't expect it to fail in the same way as a more recent one.

It's interesting that the version command works at the CLI.  One guess is something strange is happening when marimo tries to open a port on your machine.

I know this is an inconvenient thing to try, but if you haven't already tried this: can you restart your machine and try again?

**Reply by MrHenning** [reply, marked as answer]
Ha! Restarting did not change anything.
But I removed marimo again using conda, re-installed it with pip, and it still didn't work.
I removed it with pip, created a new conda env and re-installed it with conda, and now it works :smiley: 
I am sure I have done all these things before, without success...

Anyway, thanks a lot for your time @akshayka !

And btw, marimo is absolutely fantastic!!! I was missing the reactivity and interactivity in Python notebooks so much after getting used to Pluto.jl with Julia!

**Reply by akshayka** [reply]
Phew, glad you were able to figure it out! Managing installations of Python tools and environments can be quite the headache -- we've all been there.

Thanks for the kind words!

**Reply by gganapat** [reply]
I'm having the same problem as prior post. I installed marimo on my M1 Mac this morning, and running `marimo tutorial intro `does nothing - just hangs. I uninstalled and reinstalled using `python3 -m pip install marimo`. The installation seemed to go smoothly but it doesn't seem to work. I'm using Python 3.10. I tried reinstalling marimo and same thing - nothing happened. If I don't want to use conda is that a problem?

Thanks!

Gani -

**Reply by akshayka** [reply]
@gganapat 

I don't use conda, so you shouldn't have to either.

Have you accidentally installed `marimo` system-wide? You can check by typing `which marimo` (paste the output here if you need help).

Here is what usually fixes this problem.

1. Deactivate all virtual environments.
2. `pip uninstall marimo`
3. Create and activate a new virtual environment.
4. Install marimo in that virtual environment.
5. Try running marimo again. 

If `python -m marimo tutorial intro` works but `marimo tutorial intro` doesn't, that's a hint marimo may have been installed system-wide.

If this still doesn't work let me know.

**Comment by gganapat** [comment]
I figured out the problem I was having. I had an entry in my /etc/hosts for
127.0.0.1 which was non-standard instead of 'localhost' (I had it changed
for another server set up I had a long time ago). That fixed the problem. I
tried this after uninstalling every Python version installation I had along
with their symbolic links and reinstalling which didn't work.

gani -
